packages/eslint-plugin/src/rules/prefer-literal-enum-member.ts

```diff
@@ -0,0 +1,41 @@
+import { createRule } from '../util';
+import { AST_NODE_TYPES } from '@typescript-eslint/experimental-utils';
+import { TSEnumMember } from '../../../typescript-estree/dist/ts-estree/ts-estree';
+
+export default createRule<[], 'notLiteral'>({
+  name: 'prefer-literal-enum-member',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Require that all enum members be literal values to prevent unintended enum member name shadow issues',
+      category: 'Best Practices',
+      recommended: false,
+      requiresTypeChecking: false,
+    },
+    messages: {
+      notLiteral: `Enum member must be a ${AST_NODE_TYPES.Literal}.`,
```

Comment:
nit - wording, and add some examples to make it super explicit and clear

```suggestion
      notLiteral: 'Explicit enum value must only be a literal value (string, number, boolean, etc).',
```

Reply:
Sure.